File: frontend/src/types/image.ts
export interface Image {
  id: string;
  project_id: string;
  filename: string;
  original_path: string;
  thumbnail_path?: string;
  width: number;
  height: number;
  file_size?: number;
  format?: string;
  image_class?: string;
  uploaded_by?: string;
  created_at: string;
  metadata: Record<string, any>;
  annotation_count: number;
  annotation_classes: string[];
}

export interface ImageUpload {
  project_id: string;
  file: File;
}

export interface ImageUpdate {
  image_class?: string | null;
}

File: backend/app/schemas/image.py
"""Image schemas"""
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime
from typing import Optional, Dict, Any, List


class ImageUpload(BaseModel):
    """Schema for image upload"""
    project_id: UUID


class ImageUpdate(BaseModel):
    """Schema for image update"""
    image_class: Optional[str] = None


class ImageResponse(BaseModel):
    """Schema for image response"""
    id: UUID
    project_id: UUID
    filename: str
    original_path: str
    thumbnail_path: Optional[str]
    width: int
    height: int
    file_size: Optional[int]
    format: Optional[str]
    image_class: Optional[str] = None
    uploaded_by: Optional[UUID]
    created_at: datetime
    metadata: Dict[str, Any] = {}
    annotation_count: int = 0
    annotation_classes: List[str] = []

    class Config:
        from_attributes = True
